spark-operator/src/main/java/org/apache/spark/k8s/operator/SparkOperator.java

```diff
@@ -0,0 +1,251 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.spark.k8s.operator;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.javaoperatorsdk.operator.Operator;
+import io.javaoperatorsdk.operator.RegisteredController;
+import io.javaoperatorsdk.operator.api.config.ConfigurationServiceOverrider;
+import io.javaoperatorsdk.operator.api.config.ControllerConfigurationOverrider;
+import lombok.extern.slf4j.Slf4j;
+import okhttp3.Interceptor;
+import org.apache.commons.collections.CollectionUtils;
+
+import org.apache.spark.k8s.operator.client.KubernetesClientFactory;
+import org.apache.spark.k8s.operator.config.SparkOperatorConf;
+import org.apache.spark.k8s.operator.config.SparkOperatorConfigMapReconciler;
+import org.apache.spark.k8s.operator.listeners.SparkAppStatusListener;
+import org.apache.spark.k8s.operator.metrics.MetricsService;
+import org.apache.spark.k8s.operator.metrics.MetricsSystem;
+import org.apache.spark.k8s.operator.metrics.MetricsSystemFactory;
+import org.apache.spark.k8s.operator.metrics.healthcheck.SentinelManager;
+import org.apache.spark.k8s.operator.metrics.source.KubernetesMetricsInterceptor;
+import org.apache.spark.k8s.operator.metrics.source.OperatorJosdkMetrics;
+import org.apache.spark.k8s.operator.probe.ProbeService;
+import org.apache.spark.k8s.operator.reconciler.SparkAppReconciler;
+import org.apache.spark.k8s.operator.reconciler.SparkReconcilerUtils;
+import org.apache.spark.k8s.operator.utils.ClassLoadingUtils;
+import org.apache.spark.k8s.operator.utils.SparkAppStatusRecorder;
+
+/**
+ * Entry point for Spark Operator. Sets up reconcilers for CustomResource and health check servers
+ */
+@Slf4j
+public class SparkOperator {
```

Comment:
We need a proper test coverage for `SparkOperator`.

For the record, this is a dangling file in this PR. I can remove this single file and run Gradle test 
```
$ rm spark-operator/src/main/java/org/apache/spark/k8s/operator/SparkOperator.java
$ ./gradlew test
...
BUILD SUCCESSFUL in 1m 5s
14 actionable tasks: 14 executed
```